modules/core/src/com/haulmont/cuba/core/app/EntitySnapshotManager.java

```diff
@@ -336,4 +315,95 @@ protected String processSnapshotXml(String snapshotXml, Map<Class, Class> classM
         replaceInXmlTree(document.getRootElement(), classMapping);
         return document.asXML();
     }
+
+    @Override
+    public EntitySnapshot getLastEntitySnapshot(Entity entity) {
+        MetaClass metaClass = entity.getMetaClass();
```

Comment:
Get original meta class first. Do not use actual meta class, because sometimes extended entities are removed and original are used later.